pkg/egressip/egressip.go

```diff
@@ -0,0 +1,317 @@
+package egressip
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+
+	egressipv1 "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/egressip/v1"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// EgressIPBuilder provides a struct for EgressIP object.
+type EgressIPBuilder struct {
+	// EgressIP definition, used to create the EgressIP object.
+	Definition *egressipv1.EgressIP
+	// Created EgressIP object.
+	Object *egressipv1.EgressIP
+	// Used to store latest error message upon defining or mutating EgressIP definition.
+	errorMsg string
+	// api client to interact with the cluster.
+	apiClient goclient.Client
+}
+
+// NewEgressIPBuilder creates a new instance of EgressIP builder.
+func NewEgressIPBuilder(apiClient *clients.Settings, name string) *EgressIPBuilder {
+	glog.V(100).Infof(
+		"Initializing new EgressIP structure with the following params: name: %s", name)
+
+	if apiClient == nil {
+		glog.V(100).Infof("EgressIP 'apiClient' cannot be empty")
+
+		return nil
+	}
+
+	err := apiClient.AttachScheme(egressipv1.AddToScheme)
+	if err != nil {
+		glog.V(100).Infof("Failed to add 'egressip' scheme to client schemes")
+
+		return nil
+	}
+
+	builder := &EgressIPBuilder{
+		apiClient: apiClient.Client,
+		Definition: &egressipv1.EgressIP{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: name,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name parameter of the EgressIP is empty")
+
+		builder.errorMsg = "the name parameter of the EgressIP is empty"
+
+		return builder
+	}
+
+	return builder
+}
+
+// WithEgressIPs applies egressIPs to the EgressIP spec definition.
+func (builder *EgressIPBuilder) WithEgressIPs(egressIPs []string) *EgressIPBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Applying egressIPs %s to EgressIP %q", egressIPs, builder.Definition.Name)
+
+	if len(egressIPs) == 0 {
+		glog.V(100).Infof("The egressIP is empty")
+
+		builder.errorMsg = "cannot accept empty list as egressIPs value"
+
+		return builder
+	}
+
+	builder.Definition.Spec.EgressIPs = egressIPs
+
+	return builder
+}
+
+// WithNamespaceSelector applies namespaceSelector to the EgressIP spec definition.
+func (builder *EgressIPBuilder) WithNamespaceSelector(namespaceSelector metav1.LabelSelector) *EgressIPBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Applying podSelector %s to EgressIP %q",
```

Comment:
```suggestion
	glog.V(100).Infof("Applying namespaceSelector %s to EgressIP %q",
```
one more I missed last time